php_phongo.c

```diff
@@ -775,7 +849,24 @@ int phongo_execute_command(mongoc_client_t *client, const char *db, zval *zcomma
 	/* Although "opts" already always includes the serverId option, the read
 	 * preference is added to the command parts, which is relevant for mongos
 	 * command construction. */
-	if (!mongoc_client_command_with_opts(client, db, command->bson, phongo_read_preference_from_zval(zreadPreference TSRMLS_CC), opts, &reply, &error)) {
+	switch (type) {
+		case PHONGO_COMMAND_RAW:
+			result = mongoc_client_command_with_opts(client, db, command->bson, phongo_read_preference_from_zval(zreadPreference TSRMLS_CC), opts, &reply, &error);
+			break;
+		case PHONGO_COMMAND_READ:
+			result = mongoc_client_read_command_with_opts(client, db, command->bson, phongo_read_preference_from_zval(zreadPreference TSRMLS_CC), opts, &reply, &error);
+			break;
+		case PHONGO_COMMAND_WRITE:
+			result = mongoc_client_write_command_with_opts(client, db, command->bson, opts, &reply, &error);
+			break;
+		case PHONGO_COMMAND_READ_WRITE:
+			result = mongoc_client_read_write_command_with_opts(client, db, command->bson, phongo_read_preference_from_zval(zreadPreference TSRMLS_CC), opts, &reply, &error);
```

Comment:
The read preference argument is not used for RW mode, so you should be able to pass NULL.